packages/c/sshnpd/src/params.c

```diff
@@ -106,71 +138,42 @@ int parse_sshnpd_params(sshnpd_params *params, int argc, const char **argv) {
     }
   }
 
-  // malloc pointers to each string, but don't malloc any more memory for individual char storage
-  params->manager_list = malloc((sep_count + 1) * sizeof(char *)); // FIXME: leak
-  if (params->manager_list == NULL) {
-    printf("Failed to allocate memory for manager list\n");
-    free(params->permitopen_str);
-    return 1;
-  }
-  params->manager_list[0] = manager;
-  int pos = 1; // Starts at 1 since we already added the first item to the list
-  for (int i = 0; i < manager_end; i++) {
-    if (manager[i] == ',') {
-      // Set this comma to a null terminator
-      manager[i] = '\0';
-      if (manager[i + 1] == '\0') {
-        // Trailing comma, so we over counted by one
-        sep_count--;
-        // The allocated memory has a double trailing null seperator, but that's fine
-        break;
-      }
-      if (manager[i + 1] != '@') {
-        printf("Invalid Argument(s): Expected a list of atSigns: \"%s\"\n", manager);
-        free(params->manager_list);
-        free(params->permitopen_str);
-        return 1;
-      }
-      // Keep track of the start of the next item
-      params->manager_list[pos++] = manager + i + 1;
-    }
-  }
-  params->manager_list_len = sep_count + 1;
-
-  // Repeat for permit-open
-  sep_count = 0;
-  for (int i = 0; i < permitopen_end - 1; i++) {
-    if (permitopen[i] == ',') {
-      sep_count++;
+  int pos; // position counter
+  if (manager != NULL) {
+    // malloc pointers to each string, but don't malloc any more memory for individual char storage
+    params->manager_list = malloc((sep_count + 1) * sizeof(char *)); // FIXME: leak
+    if (params->manager_list == NULL) {
+      printf("Failed to allocate memory for manager list\n");
+      free(params->permitopen_str);
+      return 1;
```

Comment:
refactored the manager_list code to a cleaner solution more similar to permitopen...
I was referencing this as I was doing the permitopen parsing and realized this need to be changed.